R/data.table.R

```diff
@@ -147,6 +147,33 @@ replace_dot_alias = function(e) {
     on.exit(options(oldverbose))
   }
   .global$print=""
```

Comment:
This is very early in ```[.data.table```. Does it make sense to skip the initial arg checks? If it does, then it would be easy enough to skip arg checking until they're used for the first time.

For example, are we OK with no errors from:
```
dt[, 1:2, with = c(j = TRUE), by = whatever, .SDcols = yeah_cool]
```

Reply:
The point is to escape early, so the less checks the better. When such a `with` provided then only `i` and `j` are relevant.